servers/services/src/main/java/org/projectnessie/services/rest/TreeResource.java

```diff
@@ -335,6 +337,56 @@ public EntriesResponse getEntries(String namedRef, EntriesParams params)
     }
   }
 
+  @Override
+  public EntriesResponse getNamespaceEntries(
+      String refName,
+      @Nullable String hashOnRef,
+      @Nullable ContentsKey namespacePrefix,
+      @Nullable Integer depth)
+      throws NessieNotFoundException {
+    WithHash<NamedRef> refWithHash = namedRefWithHashOrThrow(refName, hashOnRef);
+    try {
+      List<EntriesResponse.Entry> entries;
+      try (Stream<EntriesResponse.Entry> keyStream =
+          getStore()
+              .getKeys(refWithHash.getHash())
+              .map(
+                  key ->
+                      EntriesResponse.Entry.builder()
+                          .name(fromKey(key.getValue()))
+                          .type((Type) key.getType())
+                          .build())) {
+        entries =
+            filterEntries(
+                    keyStream,
+                    (namespacePrefix == null || namespacePrefix.getElements().isEmpty())
+                        ? null
+                        : buildRegex(namespacePrefix))
+                .map(e -> truncate(e, depth))
```

Comment:
Can't we construct the regex in a way, that it takes care of depth (matches only if the number of separators= depth)? If yes, a secondary truncate isn't necessary.

Reply:
no, unfortunately. There are no namespace objects so there is no `a.b` only `a.b.table` so we always have to truncate the `table` part when searching for `a.b`

Reply:
Sorry for confusing, I meant use of Matcher.group(..) ; but this version is also fine.